Dockerfile

```diff
@@ -33,8 +33,9 @@ RUN wget -q -O - https://apt.mopidy.com/mopidy.gpg \
 # Clone Iris from the repository and install in development mode.
 # This allows a binding at "/iris" to map to your local folder for development, rather than
 # installing using pip.
-RUN git clone https://github.com/jaedb/Iris.git /iris \
- && cd /iris \
+COPY . /iris
```

Comment:
IMO this is better because it allows the local development flow to continue without first requiring a push to the (upstream) git repository. I can revert if you disagree, though @jaedb

Reply:
The original idea here is that you bind a volume to `/iris`, which would supercede what was pulled from GitHub. From memory copying the current directory was fraught with issues because this would include source files not required for running Iris, but cloning was a safe way to pull only what was required.

Reply:
I must be misunderstanding something, because AFAICT the two commands are exactly the same (except for where they get their source). `git clone` copies all files from the repo, whereas `copy .` copies the local version of the repository. The only difference in the copied files would be anything on the local directory excluded via `.gitignore` and therefore missing from the git repo, but those would not be present on a fresh pull anyways.

You can validate all that by running this in a Dockerfile:
```
RUN git clone https://github.com/jaedb/Iris.git /iris-git && ls -la /iris-git
COPY . /iris-local && ls -la /iris-local
```

They both clone into the exact same location (`/iris`) and neither of them would preclude using a volume binding to overwrite that. The way Docker works, the `/iris` directory is mounted within the built layer, but any volume binding can always overwrite that.

Reply:
Btw, if you want to do any testing of the container yourself, I've pushed it to `inzania/iris`. Happy to remove it once this ships so as to not confuse folks.

Reply:
The biggest difference is that the repo doesn't contain everything that might be in your local directory (`./`). For example the possibly sensitive `mopidy.conf` if you've got tokens/passwords configured.
Pulling from GitHub protected these from accidentally being copied into a publicly available Docker image.

Reply:
Ah, that's a good point. I didn't consider that people might modify the local config files with sensitive data; I was thinking of them as templates to be overwritten from external sources. Okay, I'll go ahead and revert this — any other steps we need to take to merge?